lib/ansiblelint/rules/PackageHasRetryRule.py

```diff
@@ -76,12 +76,31 @@ class PackageHasRetryRule(AnsibleLintRule):
         "absent",
     ]
 
+    _package_name_keys = [
+        "name",
+        "package",
+        "pkg",
+        "deb",
+    ]
+
     def matchtask(self, file, task):
         module = task["action"]["__ansible_module__"]
-        if module in self._package_modules:
-            if 'until' not in task:
-                # Exclude tasks where state is `absent`.
-                if task['action'].get('state') not in self._module_ignore_states:
-                    message = '{0} {1}'
-                    return message.format(self.shortdesc, module)
-                return False
+
+        if module not in self._package_modules or 'until' in task:
+            return False
+
+        if task['action'].get('state') in self._module_ignore_states:
+            return False
+
+        for key in self._package_name_keys:
```

Comment:
Let's start with a generator expression for extracted package to have it lazy evaluated and stop. Now, let's have a generator for filtering empty modules away. Finally, let's retrieve :
```python
packages_gen = (
    task['action'].get(key) for key in self._package_name_keys
)
non_empty_packages_gen = (
    package_name for package_name in packages_gen
    if package_name
)
try:
    package_name = next(non_empty_packages_gen)
except StopIteration:
    return True  # there's no non-empty packages
```

First two steps could be
```python
packages_gen = map(task['action'].get, self._package_name_keys)
non_empty_packages_gen = filter(bool, packages_gen)
```
if you prefer FP style but most of Pythonistas agree that it's non-pythonic.

Reply:
I agree it's not readable enough as-is, I'm thinking this approach might be best for future readers:
```
        for key in self._package_name_keys:
            package_name = task['action'].get(key)
            if package_name:
                break
        if package_name:
            is_package_file = '.' in package_name
            is_package_html = '://' in package_name
            is_local_package_file = is_package_file and not is_package_html
            if is_local_package_file:
                return False
```

Reply:
It's not a good idea to "side-effect" vars outside of for-block.
If you want to keep it as a loop block, better put it into a function returning that package name then.